src/lib/crypto/snarky_tests/snarky_tests.ml

```diff
@@ -434,6 +434,44 @@ let api_tests =
   ; ("compile monadic API", `Quick, MonadicAPI.get_hash_of_circuit)
   ]
 
+module Protocol_circuits = struct
+  let proof_level, constraint_constants =
+    Genesis_constants.(Proof_level.Full, Constraint_constants.compiled)
+
+  let blockchain () : unit =
+    let expected = "fe5c73e8189b27aa30ea96076825416e" in
+
+    let digest =
+      Blockchain_snark.Blockchain_snark_state.constraint_system_digests
+    in
+    let digest = digest ~proof_level ~constraint_constants () in
+    let _, hash = List.hd_exn digest in
+    let digest = Md5.to_hex hash in
+
+    Format.printf "expected:\n%s\n\n" expected ;
+    Format.printf "obtained:\n%s\n" digest ;
+    assert (String.(digest = expected)) ;
```

Comment:
maybe only print if the digests differ instead of always printing?

Reply:
Done, I should mention that they already only appear when it fails, I don't know if it's being captured or what causes it.